bigtable/use-cases/fraudDetection/README.md

```diff
@@ -0,0 +1,87 @@
+# Credit card fraud-detection using Cloud Bigtable
+
+This sample application aims to build a fast and scalable fraud detection system using Cloud Bigtable as its feature store. The feature store holds Demographic information (customer ids, addresses, etc) and Historical transactions. When deciding if a transaction is fraudulent or not, the feature store is queried to fetch the customer demographic information and the transaction history.
+
+Cloud Bigtable is a great fit to use as a feature store for the following reasons:
+
+1.  **Scalable:** Cloud Bigtable can handle petabytes of data allowing the fraud-detection service to scale to many customers.
+    
+2.  **Fast:** It has a very low latency which helps in this use case because the system needs to identify if a transaction is fraudulent or not as soon as possible.
+    
+3.  **Low maintenance:** Lower maintenance compared to relational offerings. Also, Cloud Bigtable supports garbage collection policies which simplify customers’ history management.
+
+## System design
+![Fraud detection design](fraud-detection-design.jpg)
+
+**1.  Input/Output Cloud Pubsub topics:** The real-time transactions arrive at the Cloud Pubsub input topic, and the output is sent to Cloud Pubsub output topic.
+    
+**2.  ML Model:** The component that decides the probability of a transaction of being fraudulent. This project provides a pre-trained ML model and hosts it on VertexAI (See ML Model section).
+    
+**3.  Cloud Bigtable as a Feature Store:** Cloud Bigtable stores Customers’ demographics and historical data. The Dataflow pipeline queries Cloud Bigtable in real-time when to get and aggregate the customer’s demographics and historical data.
+    
+**4.  Dataflow Pipeline:** The streaming pipeline that orchestrates this whole operation. It reads the transaction details from the Cloud Pubsub input topic, queries Cloud Bigtable to build a feature vector that is sent to the ML model, and lastly, it writes the output to the Cloud Pubsub output topic.
+    
+**5.  Data warehouse (BigQuery, Spark, etc):** This component stores the full history of all transactions queried by the system. It runs batch jobs for continuously training the ML model. Note that this component is outside the scope of this project as a pre-trained ML model is provided for simplicity.
+
+The system design is written using the Terraform framework. All components details can be found in the file **terraform/main.tf** and it includes the components listed above.
+
+## Datasets
+
+This project uses [Sparkov Data Generation Simulator](https://github.com/namebrandon/Sparkov_Data_Generation) to generate the datasets that are used for training the ML model and for testing it.
```

Comment:
I had to change some names in that generator and perform some cleanup to make it work with our ML model. I don't think it will be easy to do that on the go. That being said, that's something we could improve in the future.

Reply:
Okay!